apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabTheorycraft/GcsimButton.tsx

```diff
@@ -0,0 +1,167 @@
+import type { MainStatKey, SubstatKey } from '@genshin-optimizer/consts'
+import { ascensionMaxLevel, getMainStatValue } from '@genshin-optimizer/gi-util'
+import { useBoolState } from '@genshin-optimizer/react-util'
+import { CardThemed, CodeBlock } from '@genshin-optimizer/ui-common'
+import { toDecimal } from '@genshin-optimizer/util'
+import ContentPasteIcon from '@mui/icons-material/ContentPaste'
+import {
+  Button,
+  CardContent,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  Link,
+} from '@mui/material'
+import { useContext } from 'react'
+import { Trans, useTranslation } from 'react-i18next'
+import ImgIcon from '../../../../Components/Image/ImgIcon'
+import { CharacterContext } from '../../../../Context/CharacterContext'
+import { CharTCContext } from './CharTCContext'
+import gcsimIcon from './gcsim.png'
+
+// From https://github.com/genshinsim/gcsim/blob/68874fb693c4e959b336d106f201c29d100fc2c8/pkg/core/attributes/stats.go#L72
+const GOODtoSRL: Record<MainStatKey | SubstatKey, string> = {
```

Comment:
Is it worth refactoring this const + the converter logic out to a lib now? gi-gcsim or something? go-next will want this at some point i'm sure

Reply:
Probably when go-next get to this stage.

Reply:
If we plan to do it in the future, why not do it now and avoid the tech debt and refactor pr?

Reply:
because I don't plan to do it in the future, I am not even sure I'll keep supporting gcsim.

Reply:
If we don't want to, or plan to support it in the future, then I would say we shouldn't add it in the first place

Reply:
Because I don't plan for it doesn't mean that it won't happen in the future. This is a ask mostly from the KQM side, and I want to see what happens in terms of feedback/opportunity for evolution before I commit too hard into gcsim support.